tests/testthat/helper-preload.R

```diff
@@ -19,6 +19,22 @@ ap <- function(.name, .prefix) {
   paste0(.prefix, .name)
 }
 
+add_mock_sample <- function(.immdata, .sample_name, .meta = NA) {
+  # copy dataframe of 1st sample to the new sample
+  .immdata$data[[.sample_name]] <- .immdata$data[[1]]
+  .immdata$meta %<>% add_row(Sample = .sample_name)
+  if (!is.na(.meta)) {
+    # .meta must be a list of lists with 2 elements: meta_column and meta_value
+    for (i in seq_along(.meta)) {
+      meta_column <- .meta[[i]][["meta_column"]]
+      meta_value <- .meta[[i]][["meta_value"]]
+      # set meta value in the last (just added) row
+      .immdata$meta[nrow(.immdata$meta), meta_column] <- meta_value
+    }
+  }
```

Comment:
if meta is named list (which is essential as in repFilter realisation) you can use `bind_rows(.immdata$meta, as.data.frame(.meta))` instead of lines 27-33. It will helps to prevent multiple indexing  and using a lot of wrapped lists.